### PR TITLE
YLP-858 Some algorithm settings are visible on the daily view

### PR DESCRIPTION
**Pull request by CedricPerotto**

### CODE REVIEW

**Review by Nas84 — APPROVED** [review]
LGTM

**Review by yvon-dblg — COMMENTED** [review]
I would probably have created statically the levels arrays.
To avoid allocation at each run and conversion for each parameter from int to string.
(Not tested yet)

```golang
var (
	// Parameters level to keep in api response (int value)
	paramLevelFilterInt = [...]int{1}
	// Parameters level to keep in api response (string value)
	paramLevelFilterStr = [...]string{"1"}
)
```

Call to the GetDiabeloopParametersHistory:
```golang
		writeParams.parametersHistory, err = a.store.GetDiabeloopParametersHistory(ctx, userID, paramLevelFilterInt[:])
```

Filter for the daily view:
```golang
			if datumType == "deviceEvent" {
				datumSubType, haveSubType := datum["subType"].(string)
				if haveSubType && datumSubType == "deviceParameter" {
					datumLevel, haveLevel := datum["level"].(string)
					if haveLevel && !contains(paramLevelFilterStr[:], datumLevel) {
						continue
					}
				}
			}
```

File: data/api_v1.go [hunk 1]
```diff
@@ -189,6 +192,10 @@ func (a *API) getDataV1(ctx context.Context, res *httpResponseWriter) error {
 		uploadIDs: make([]string, 0, 16),
 	}
 
+	parameterLevelFilter := make([]int, 1)
+	parameterLevelFilter = append(parameterLevelFilter, 1)
```
Comment (franck-fourel) [inline data/api_v1.go:196]:
```suggestion
	parameterLevelFilter := [2]int{1, 2}
```

File: data/api_v1_test.go [hunk 2]
```diff
@@ -101,3 +101,54 @@ func TestAPI_GetDataV1(t *testing.T) {
...
+	}
+
+	body := make([]byte, 1024)
+	defer result.Body.Close()
+	n, _ := result.Body.Read(body)
+	bodyStr := string(body[:n])
+
+	expectedBody := `[
+{"id":"01","level":"0","subType":"deviceParameter","time":"2021-01-10T00:00:00.000Z","type":"deviceEvent","uploadId":"00"},
+{"id":"02","level":"1","subType":"deviceParameter","time":"2021-01-10T00:00:01.000Z","type":"deviceEvent","uploadId":"00"},
```
Comment (franck-fourel) [inline data/api_v1_test.go:147]:
```suggestion
{"id":"01","level":"1","subType":"deviceParameter","time":"2021-01-10T00:00:01.000Z","type":"deviceEvent","uploadId":"00"},
{"id":"02","level":"2","subType":"deviceParameter","time":"2021-01-10T00:00:02.000Z","type":"deviceEvent","uploadId":"00"},
```

File: data/api_v1_test.go [hunk 3]
```diff
@@ -101,3 +101,54 @@ func TestAPI_GetDataV1(t *testing.T) {
...
+	traceID := uuid.New().String()
+	userID := "abcdef"
+	urlParams := map[string]string{
+		"userID": userID,
+	}
+
+	storage.DataV1 = []string{
+		"{\"id\":\"01\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:00.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"0\"}",
+		"{\"id\":\"02\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:01.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"1\"}",
+		"{\"id\":\"03\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:02.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"2\"}",
```
Comment (franck-fourel) [inline data/api_v1_test.go:115]:
```suggestion
		"{\"id\":\"01\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:01.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"1\"}",
		"{\"id\":\"02\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:02.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"2\"}",
		"{\"id\":\"03\",\"uploadId\":\"00\",\"time\":\"2021-01-10T00:00:03.000Z\",\"type\":\"deviceEvent\",\"subType\":\"deviceParameter\",\"level\":\"3\"}",```

**Review by franck-fourel — CHANGES_REQUESTED** [review]
some little changes based on what devdroid specifications

**Review by franck-fourel — APPROVED** [review]

**Pull request merged**